bivrost-abi-parser/src/main/kotlin/pm/gnosis/AbiParser.kt

```diff
@@ -30,12 +92,61 @@ class AbiParser {
             build.writeTo(output)
         }
 
+        private fun checkType(type: String): String {
+            return Solidity.aliases.getOrElse(type, { type })
+        }
+
+        internal fun mapType(type: String): TypeHolder {
+            // uint[5][]
+            val matcher = TYPE_PATTERN.matcher(type)
+            matcher.find()
+            val arrayType = matcher.group(1)
+            val baseType = Solidity.types[checkType(arrayType)] ?: throw IllegalArgumentException("Unknown type $type!")
+            val arrayDef = matcher.group(2)
+            return parseArrayDefinition(arrayDef, SimpleTypeHolder(ClassName.bestGuess(baseType), isSolidityDynamicType(arrayType)))
+        }
+
+        private fun parseArrayDefinition(arrayDef: String, innerType: TypeHolder): TypeHolder {
+            if (arrayDef.isBlank()) {
+                return innerType
+            }
+            val length = arrayDef.length
+            if (length < 2 || arrayDef[0] != '[') {
+                // Missing opening bracket or too short
+                throw IllegalArgumentException("Illegal array definition $arrayDef!")
+            }
+
+            var closingBracketIndex = -1
+            for (i in 1 until length) {
+                val c = arrayDef[i]
+                if (c == ']') {
+                    closingBracketIndex = i
+                    break
+                } else if (!c.isDigit()) {
+                    // Size of array not numeric
+                    throw IllegalArgumentException("Illegal array definition $arrayDef!")
+                }
+            }
+            if (closingBracketIndex < 1) {
+                // Missing closing bracket
+                throw IllegalArgumentException("Illegal array definition $arrayDef!")
+            }
```

Comment:
We could, but it would not validate all cases. E.g `uint[[][9]` would be valid.

also the pattern should probably be `"\\[[0-9]*]"`. And we should have a Pattern object in the companion object.

Reply:
actually I am not sure what happens if you would use that invalid definition right now ... might be parsed into a `uint`